services/net/notification/NotificationManager.cs

```diff
@@ -432,46 +432,72 @@ private async Task SendNotificationAsync(
         await HandleChesEmailOverrideAsync(request);
 
         var subscribers = await GetNotificationSubscribersAsync(notification, content);
-        var contexts = new List<EmailContextModel>();
+
+        var body = await GenerateNotificationBodyAsync(notification, content, null, request.IsPreview); 
+        var subject = string.Empty;
         if (!String.IsNullOrWhiteSpace(request.To))
-        {
+        {           
+            var contexts = new List<EmailContextModel>();
             // When a notification request has specified 'To' it means only send it to the emails in that property.
             var requestTo = request.To.Split(",").Select(v => v.Trim());
             contexts.Add(new EmailContextModel(requestTo, new Dictionary<string, object>(), DateTime.Now));
+            // There are no subscribers, or a notification has been sent for this content to all the subscribers.
+            if (!contexts.Any())
+            {
+                if (!subscribers.Any())
+                {
+                    this.Logger.LogDebug("Notification does not have subscribers. Notification: {notificationId}", notification.Id);
+                }
+                else
+                {
+                    this.Logger.LogInformation("Notification is not sent because all users has already received this content. Notification: {notificationId}", notification.Id);
+                }
+                return;
+            }
+            subject = await GenerateNotificationSubjectAsync(notification, content, request.IsPreview, false);
+            SendOutNotificationEmailsAsync(contexts, subject, body, new List<API.Areas.Services.Models.Notification.UserModel>(), request, notification, content);
         }
         else
         {
-            contexts.AddRange(this.NotificationValidator.GetSubscriberEmails(subscribers));
-        }
+            foreach(var subscriber in subscribers)
```

Comment:
Are you sending out a separate email to each subscriber now?  If so, you shouldn't use the mail merge feature.